Yoast/wordpress-seo
# P2 364 block validation warnings

## Context


* Adds sidebar analysis warnings explaining exactly what is 'wrong' with the user's post.

## Summary


This PR can be summarized in the following changelog entry:

* Adds sidebar warnings that explain schema block validations.

## Relevant technical choices:

* changed the way validations are stored for much easier retrieval of validations, by clientid.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Add a job posting. 
* See the sidebar; there should be an Analysis header containing some complaints about missing or empty values.
* Add a paragraph with some text.
* Publish the post; view the published post's source code; there should NOT be schema for the job posting, but there should be schema for the other (valid!) elements on the page, e.g. the paragraph.

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Start a new post, Add a block that has some requirements, such as a job block.
* the sidebar should show red bullets explaining what's wrong, followed by a message stating there won't be any schema for this block until the issues are fixed.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes #
